pkg/testcase/selinux.go

```diff
@@ -0,0 +1,106 @@
+package testcase
+
+import (
+    "fmt"
+	"github.com/rancher/distros-test-framework/pkg/assert"
+	"github.com/rancher/distros-test-framework/factory"
+	"github.com/rancher/distros-test-framework/shared"
+
+	. "github.com/onsi/ginkgo/v2"
+)
+
+
+// Validate that containerd is running with selinux enabled in the config
+func TestSelinuxEnabled(product string) {
+	ips := shared.FetchNodeExternalIP()
+	selinuxConfigAssert := "selinux: true"
+	selinuxContainerdAssert := "enable_selinux = true"
+
+	for _, ip := range ips {
+		assert.CheckComponentCmdNode("cat /etc/rancher/"+product+"/config.yaml", ip, selinuxConfigAssert)
+		assert.CheckComponentCmdNode("sudo cat /var/lib/rancher/"+product+"/agent/etc/containerd/config.toml", ip, selinuxContainerdAssert)
+	}
+}
+
+// Validate container-selinux version, rke2-selinux version and rke2-selinux version
+func TestSelinuxVersions(product string) {
+	cluster := factory.GetCluster(GinkgoT())
+
+	var serverAsserts [] string
+
+	if product == "rke2"{
+	    serverAsserts = []string{"container-selinux", product + "-selinux", product + "-server"}
+	}else{
+	    serverAsserts = []string{"container-selinux", product + "-selinux"}
+	}
+
+	agentAsserts := []string{"container-selinux", product + "-selinux"}
+
+	if cluster.NumServers > 0 {
+		for _, serverIP := range cluster.ServerIPs {
+
+		    if product == "rke2"{
+		        assert.CheckComponentCmdNode("rpm -qa container-selinux " + product + "-server " + product + "-selinux", serverIP, serverAsserts...)
+		    }else{
+		        assert.CheckComponentCmdNode("rpm -qa container-selinux " + product + "-selinux", serverIP, serverAsserts...)
+		    }
+		}
+	}
+
+	if cluster.NumAgents > 0 {
+		for _, agentIP := range cluster.AgentIPs {
+			assert.CheckComponentCmdNode("rpm -qa container-selinux " + product + "-selinux", agentIP, agentAsserts...)
+		}
+	}
+}
+
+// Validate directories to ensure they have the correct selinux contexts created
+
+// Validate that containers don't run with spc_t
+func TestSelinuxSpcT() {
+	cluster := factory.GetCluster(GinkgoT())
+	securityAssert := "spc_t"
+
+	for _, serverIP := range cluster.ServerIPs {
+		assert.CheckNotPresentOnNode("ps auxZ | grep metrics | grep -v grep", serverIP, securityAssert)
+		break
+	}
+}
+
+// Validate that uninstallation will remove the rke2-selinux or k3s-selinux policy
+func TestUninstallPolicy(product string) {
+    cluster := factory.GetCluster(GinkgoT())
+
+	var asserts [] string
+
+    if product == "rke2"{
```

Comment:
These few lines are duplicated... if possible create a local func to make it reusable?

Reply:
good point ....it should be better create a func to handle this , and also remove the product parameter.